### PR TITLE
Unit Testing for FeedbackUtils (TC #2 Testing)

### PR DESCRIPTION
**Pull request by Calebc4554**
# Description  
- **What does this PR do?**  
  Adds full-coverage unit testing for `feedbackUtils.js`. Tests cover location tagging, Euclidean calc, Supabase writes/reads, K-Means clustering, vector builders, and the final `recommendEventsForUser` flow.  
- **Why is it necessary or valuable?**  
  Locks down core event-recommend logic, keeps future refactors safe, and gives us quick feedback when DB shape or helper code changes.  
- **What is intentionally left out and will be done in a later PR?**  
  Testing for TC 1

---
# Milestones / Related Work  
- **Milestone / version target:**- [4.13 Unit Testing (TC #2)](https://docs.google.com/document/d/1gKQkyru2JwfG13o2M-rWdAOgh8Qr6j2nxwepipoh564/edit?tab=t.0#bookmark=id.o4mcrkpz2339)

---

# Resources and References  
- Vitest mock docs  
  https://vitest.dev/guide/mocking.html  
- Supabase JS client ref  
  https://supabase.com/docs/reference/javascript  

---

# Test Plan  
- Ran `npm test` → all suites green, coverage 100 % for `feedbackUtils.js`  
Screenshots:
image
- Seed / mock data: tiny arrays inside the test; fake Supabase client in `vi.mock`

## Edge Cases Tested  
- Unknown location → `loc:other`  
- Empty feedback map / empty vecs  
- Re-liking an already seen event  
- K-Means request with `k > vecs.length`

---

# Additional Notes  
- **Known limitations:** tests rely on hard-coded sample rows; adjust if schema changes.  
- **Planned follow-up PRs:** add mentor-matching tests.  
- **Reviewer requests:** scan comments for clarity and suggest any missing edge cases.

### CODE REVIEW

File: frontend/src/__tests__/feedbackUtils.test.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+// mock building list so getLocationCategory works
+vi.mock("../components/constants/academicBuildings", () => ({
+	academicBuildings: ["thomas building", "boucke building", "hub"],
+}));
+
+// helper function tests
+describe("helpers", () => {
+	it("getLocationCategory maps strings to tags", () => {
+		expect(getLocationCategory("Zoom link")).toBe("loc:virtual");
```
Comment (alegleason) [inline frontend/src/__tests__/feedbackUtils.test.js:44]: Maybe we should out logic held to work around an actual Zoom link e.g., https://yourcompany.zoom.us/j/1234567890 - maybe we can add some validations on the url components to check they're officially Penn State.

Comment (Calebc4554) [reply frontend/src/__tests__/feedbackUtils.test.js:44]: Implemented - Validated in add event modal.
